rxlifecycle/src/main/java/com/trello/rxlifecycle/TakeUntilGenerator.java

```diff
@@ -21,19 +22,37 @@ public Boolean call(T lifecycleEvent) {
 
     @NonNull
     @CheckResult
-    static <T> Observable<Boolean> takeUntilCorrespondingEvent(@NonNull final Observable<T> lifecycle,
-                                                               @NonNull final Func1<T, T> correspondingEvents) {
+    static <T> Observable<Boolean> takeUntilCorrespondingEvent(
+            @NonNull final Observable<T> lifecycle,
+            @NonNull final Func1<T, T> correspondingEvents) {
         return Observable.combineLatest(
-            lifecycle.take(1).map(correspondingEvents),
-            lifecycle.skip(1),
-            new Func2<T, T, Boolean>() {
-                @Override
-                public Boolean call(T bindUntilEvent, T lifecycleEvent) {
-                    return lifecycleEvent.equals(bindUntilEvent);
+                lifecycle
+                        .take(1)
+                        .map(correspondingEvents)
+                        .ambWith(Observable.<T>just(null))
```

Comment:
It's kind of enforced already. It happens to work fine for activities and fragments because people don't usually hook in prior to the constructor, and it happens to work for views because we don't map a lifecycle and just wait for a detach event. It breaks down anywhere you have a constructor that occurs before any lifecycle events (controller, presenter, what have you), and the result is it hangs trying to take(1) on a stream that has no events.